src/length-value.js

```diff
@@ -50,13 +50,25 @@
   LengthValue.prototype.add = function(addedLength) {
     if (this instanceof SimpleLength && addedLength instanceof SimpleLength && this.type == addedLength.type) {
       return this._addSimpleLengths(addedLength);
-    } 
+    } else {
+      //ensure both lengths are of type CalcLength before adding
+      var lengthToAdd_1 = this._convertToCalcLength();
+      var lengthToAdd_2 = addedLength._convertToCalcLength();
+
+      return lengthToAdd_1._addCalcLengths(lengthToAdd_2);
```

Comment:
If you change _convertToCalcLength to _asCalcLength, it's fine to just write

return (this._asCalcLength())._addCalcLengths(addedLength._asCalcLength());